exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/HashTableConfig.java

```diff
@@ -29,27 +29,58 @@
 public class HashTableConfig  {
 
   private final int initialCapacity;
+  private final boolean initialSizeIsFinal;
   private final float loadFactor;
   private final List<NamedExpression> keyExprsBuild;
   private final List<NamedExpression> keyExprsProbe;
   private final List<Comparator> comparators;
 
   @JsonCreator
-  public HashTableConfig(@JsonProperty("initialCapacity") int initialCapacity, @JsonProperty("loadFactor") float loadFactor,
+  public HashTableConfig(@JsonProperty("initialCapacity") int initialCapacity,
+                         @JsonProperty("loadFactor") float loadFactor,
                          @JsonProperty("keyExprsBuild") List<NamedExpression> keyExprsBuild,
                          @JsonProperty("keyExprsProbe") List<NamedExpression> keyExprsProbe,
                          @JsonProperty("comparators") List<Comparator> comparators) {
     this.initialCapacity = initialCapacity;
+    this.initialSizeIsFinal = false;
     this.loadFactor = loadFactor;
     this.keyExprsBuild = keyExprsBuild;
     this.keyExprsProbe = keyExprsProbe;
     this.comparators = comparators;
   }
 
+  @JsonCreator
+  public HashTableConfig(@JsonProperty("initialCapacity") int initialCapacity,
+                         @JsonProperty("initialCapacity") boolean initialSizeIsFinal,
```

Comment:
Is this constructor used with Json? If yes, how it works with `initialSizeIsFinal`bound to the same json `initialCapacity` property?

Reply:
Only one constructor should assign variables, the second constructor should use `this()` with parameters.

Reply:
Yes, that one looks like an error. And yes, we should reuse the "more comprehensive" constructor. Does this work call for a special Jira, or could it be lumped together with other expected changes ?

Reply:
IMO, it is OK to include code changes into another HashJoin PR.